File: backend/src/main/scala/microservices/feedback/objects/CreateFeedbackComplaintPlannerRequest.scala
// CreateFeedbackComplaintPlannerRequest：feedback 域后端对象定义。
package com.typesafe.travel.feedback.domain
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class CreateFeedbackComplaintPlannerRequest(
    sourceThreadId: String,
    selectedMessageIds: List[String],
    userExplanation: String,
    userDisplayName: String
)
object CreateFeedbackComplaintPlannerRequest:
  given sourceEncoder: Encoder[CreateFeedbackComplaintPlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[CreateFeedbackComplaintPlannerRequest] = deriveDecoder

File: frontend/src/microservices/feedback/objects/CreateFeedbackComplaintPlannerRequest.ts
// 本文件定义 feedback 模块的 `CreateFeedbackComplaintPlannerRequest`，用于提交投诉创建所需的线程来源、选中消息和用户说明。

export type CreateFeedbackComplaintPlannerRequest = {
  sourceThreadId: string
  selectedMessageIds: string[]
  userExplanation: string
  userDisplayName: string
}